### PR TITLE
Removes stowaways from Manifest.

### PR DESCRIPTION
**Pull request by vode-code**
[FEATURE] [BALANCE]
## About the PR 
This PR adds a check to joining and changing name that disables manifest and all other NT data reference entries when stowaway; the manifest entry for being clone scanned remains.

## Why's this needed? 
Stowaways being on the manifest is odd and impedes the true stowaway experience.

## Changelog

```changelog
(u)Chickenish
(+)Stowaways are no longer in the manifest.
```

### CODE REVIEW

**Comment by Carbadox** [comment]
Late joining as a stowaway also makes a late join annoucement. Any possibilty of making that not the case?

File: code/procs/jobprocs.dm [hunk 1]
```diff
@@ -354,7 +354,7 @@
 		src = possible_new_mob // let's hope this breaks nothing
 
 
-	if (ishuman(src) && JOB.add_to_manifest)
+	if (ishuman(src) && JOB.add_to_manifest && !src?.traitHolder.hasTrait("immigrant"))
```
Comment (ZeWaka) [inline code/procs/jobprocs.dm:357]:
```suggestion
	if (ishuman(src) && JOB.add_to_manifest && !src.traitHolder.hasTrait("immigrant"))
```
would've failed LONG before this

**Comment by vode-code** [comment]
> 
> 
> Late joining as a stowaway also makes a late join annoucement. Any possibilty of making that not the case?

from what I can tell, this is not the case.
replication steps?

**Review by ZeWaka — APPROVED** [review]
lgtm

**Pull request merged**